hail/src/main/scala/is/hail/expr/types/physical/PInt32.scala

```diff
@@ -18,6 +18,8 @@ class PInt32(override val required: Boolean) extends PIntegral {
   def _asIdent = "int32"
   def _toPretty = "Int32"
 
+  override type NType = this.type
```

Comment:
doesn't this bind `NType` to `PInt32`? Why not just write that?

Reply:
Was having typing issues, think the thing I changed it to is simpler now.